### ISSUE TITLE
How to refresh the token after the 12 days of expire

### ISSUE THREAD

**Issue by dmocas** [awinogrodzki/next-firebase-auth-edge#111]
Hello, I'm sorry if this sounds like a noob question, but I'm not understanding how can I update the token after it is expired.
I have been using the package with no problems, but after the 12th day, the token has expired and now getTokens() will always return null.

I am not using custom-claims since there are no differences between users and the examples (correct me if I'm wrong) always use the custom-claims.

I've tried implementing the middleware with no success, so I reverted it back to what I had before:
middleware.ts
` handleValidToken: async ({ token, decodedToken }, headers) => {
      
      if (PUBLIC_PATHS.includes(request.nextUrl.pathname)) {
        return redirectToHome(request);
      }

      return NextResponse.next({
        request: {
          headers,
        },
      });
    },`
    
/api/animal/all:
`export async function GET(request: NextRequest) {
  const tokens = await getTokens(request.cookies, authConfig); `

I can see that the browser cookies have the expires/Max-age setting updated to 12 days from the moment the user logs in, but the AuthToken is always the same.

What could I be doing wrong?
Thanks if advance

**Comment by dmocas** [comment]
FYI The problem only appears in production, on localhost everything works as expected

**Comment by dmocas** [comment]
![header](https://github.com/awinogrodzki/next-firebase-auth-edge/assets/121816076/08763c6d-598f-4500-9d56-83c443fe2476)
In production

![localhost](https://github.com/awinogrodzki/next-firebase-auth-edge/assets/121816076/1e31ef52-2471-4a35-9d32-31ec19eaaad8)

Localhost

**Comment by awinogrodzki** [comment]
Hey @dmocas!

Usually the token is valid for an hour. After this time, if user requests the page again, the token is automatically refreshed behind the scenes.

If `getTokens` returns null for you after 12 days, it means that Google has invalidated the certificate this users token has been signed with.

Every token is signed with Googles public certificate loaded from here: https://www.googleapis.com/robot/v1/metadata/x509/securetoken@system.gserviceaccount.com

This certificate has a expiry date, which can be deduced by `Cache-Control` header returned in it's response:

```
Cache-Control:
public, max-age=23345, must-revalidate, no-transform
```

Usually, this certificate is valid for 16 days. If that time has passed, you need to redirect the user to `/login` page and let them authenticate again.

I guess this is a security measure from Google not to allow endless access for given token and force users to eventually re authenticate

**Comment by dmocas** [comment]
Hi @awinogrodzki!
Thank you for the insight, it must have been some mistake on my part.

The problem solved itself eheh
Like you said, all that was needed was to relog. I think I must have forgotten to logout and relog on the production env., because after I read your comment, I logged out, logged in, and that's that :)

I watched as the token was deleted and replaced with a brand new one.
Didn't have a problem since, so I guess this issue can be closed.

Thank you very much!
